lib/cli/user.js

```diff
@@ -0,0 +1,43 @@
+const logger = require('../helpers/logger')
+const MongoClient = require('../core/clients/mongodb-client')
+const UserCommands = require('../core/commands/user-commands')
+
+exports.command = 'users'
+
+exports.describe = 'Sets up users against any mongo instance'
+
+exports.builder = {
+}
+
+//  MongoError {
+//   code: 11000,
+//   codeName: 'DuplicateKey',
+//   ok: 0,
+//   message: 'User "david@admin" already exists',
+// }
+
+exports.handler = async (argv) => {
+  logger.debug('Setting up users')
+  logger.debug(argv)
+
+  const { uri } = argv.connection
+  const { users } = argv
+  const client = new MongoClient(uri)
+  await client.connect()
+  const userCommands = new UserCommands(client)
```

Comment:
(This is obviously wip, but leaving a comment to be thorough)

This should be a `UserState` instead that has a `apply` function and wraps this

Reply:
Agreed this was just a hack to prove it and make it work and get feedback on naming and structure.